Necessary conditions for observability (#523)
PowerGridModel/power-grid-model: docs/user_manual/calculations.md

In observable systems this helps better outputting correct results. On the other hand with unobservable systems, exceptions raised from calculations due to faulty results will be prevented.
```

> ##### Necessary observability condition

Reviewer: i think this is the same as the above documentation (line 51-81 of this file on this branch)

Author: That explanation is from theory POV. The assumption and combining of sensor is not mentioned.

> As per the requirements of observability mentioned above, user needs to satisfy atleast  following conditions for state estimation calculation in power-grid-model. 

Reviewer: ```suggestion
As per the requirements of observability mentioned above, user needs to satisfy at least the following conditions for state estimation calculation in power-grid-model. 
```

- `n_voltage_sensor >= 1`

> - If no voltage phasor sensors are available, then following conditions should be satisfied:  `n_unique_power_sensor >= n_bus - 1`. Otherwise: `n_unique_power_sensor + n_voltage_sensor_with_phasor >= n_bus`

Reviewer: ```suggestion
- If no voltage phasor sensors are available, then the following conditions should be satisfied: `n_unique_power_sensor >= n_bus - 1`. Otherwise: `n_unique_power_sensor + n_voltage_sensor_with_phasor >= n_bus`
```

`n_unique_power_sensor` can be calculated as follows:

> - Zero injection present (ie. zero constraint for node)

Reviewer: ```suggestion
- Zero injection present (i.e. zero constraint for node)
```

- Complete injection at node: All appliances in a node are measured or a node injection sensor is present. Either of them counts as one.
- A branch sensor: Parallel branches with either side of measurements count as one.

> - A Branch3 sensor

Reviewer: how much do the zero injection and branch3 sensor contribute each?

Author: Zero injection can be only 1 on a node. 
Since branch3 is a Y form of branches, parallel branch3 and from/to side all get counted.
